src/metabase/pulse/render.clj

```diff
@@ -85,24 +85,26 @@
         (#{:pin_map :state :country} display-type)
         (chart-type nil "display-type is %s" display-type)
 
-        (#{:area
-           :bar
-           :combo
-           :funnel
-           :progress
-           :table
-           :waterfall} display-type)
-        (chart-type display-type "display-type is %s" display-type)
-
         ;; for scalar/smartscalar, the display-type might actually be :line, so we can't have line above
-        (= @col-sample-count @row-sample-count 1)
+        (and (not= display-type :progress)
+             (= @col-sample-count @row-sample-count 1))
         (chart-type :scalar "result has one row and one column")
 
         (and (= display-type :smartscalar)
              (= @col-sample-count 2)
              (seq insights))
         (chart-type :smartscalar "result has two columns and insights")
 
+        (#{:line
```

Comment:
I moved this set of keys below the `:smartscaler` and `:scalar` conditions so that I could keep `:line` inside the set. I have a separate PR that has more extensive removals of non-combo code for line, area, and bar charts, and this conditional can be simplified in that PR even further.